File: Mobile/PlatziFlixAndroid/app/src/main/java/com/espaciotiago/platziflixandroid/data/mappers/RatingMapper.kt
package com.espaciotiago.platziflixandroid.data.mappers

import com.espaciotiago.platziflixandroid.data.entities.RatingResponseDTO
import com.espaciotiago.platziflixandroid.domain.models.CourseRating

object RatingMapper {
    fun fromDTO(dto: RatingResponseDTO): CourseRating {
        return CourseRating(
            id = dto.id,
            courseId = dto.courseId,
            userId = dto.userId,
            rating = dto.rating
        )
    }
}

File: Mobile/PlatziFlixiOS/PlatziFlixiOS/Data/Mapper/RatingMapper.swift
import Foundation

struct RatingMapper {
    static func toDomain(_ dto: RatingResponseDTO) -> CourseRating {
        return CourseRating(
            id: dto.id,
            courseId: dto.courseId,
            userId: dto.userId,
            rating: dto.rating
        )
    }
}
